src/aspire/utils/numeric/pyfftw_fft.py

```diff
@@ -1,103 +1,29 @@
-from threading import Lock
+import pyfftw.interfaces.scipy_fftpack as pyfft
 
-import numpy as np
-import pyfftw
 
-mutex = Lock()
+def replace_arguments(f):
+    """
+    pyfftw uses the argument 'threads' instead of scipy.fft's 'workers' argument
+    Passing in a -ve value will allow both libraries to choose the max. number of threads as per cpu_count()
+    This decorator replaces any argument passed in as 'workers' with the argument 'threads' to make it work with
+    pyfftw.
+    :param f: The function to be decorated
+    :return: The decorated function with the 'workers' argument replaced with 'threads'
+    """
+    def inner(*args, **kwargs):
+        kwargs['threads'] = kwargs.pop('workers', None)
+        return f(*args, **kwargs)
+    return inner
 
 
 class PyfftwFFT:
 
-    # no fftshift and ifftshift from pyffw,use those from numpy
-    fftshift = staticmethod(np.fft.fftshift)
-    ifftshift = staticmethod(np.fft.ifftshift)
+    fft = staticmethod(replace_arguments(pyfft.fft))
+    ifft = staticmethod(replace_arguments(pyfft.ifft))
+    fft2 = staticmethod(replace_arguments(pyfft.fft2))
+    ifft2 = staticmethod(replace_arguments(pyfft.ifft2))
+    fftn = staticmethod(replace_arguments(pyfft.fftn))
+    ifftn = staticmethod(replace_arguments(pyfft.ifftn))
 
-    @staticmethod
-    def fft(a, axis=-1):
-        mutex.acquire()
-
-        try:
-            a_ = pyfftw.empty_aligned(a.shape, dtype='complex128')
-            b = pyfftw.empty_aligned(a.shape, dtype='complex128')
-            cls = pyfftw.FFTW(a_, b, axes=(axis,), direction='FFTW_FORWARD')
-            cls(a, b)
-        finally:
-            mutex.release()
-
-        return b
-
-    @staticmethod
-    def fft2(a, axes=(0, 1)):
```

Comment:
I guess these default arguments are never used since changing them doesn't break any tests?

Reply:
For 2D array it is fine. But it was switched to (-2, -1) to make consistent for all three methods in recent commits.